lib/backpex/resource.ex

```diff
@@ -270,8 +270,15 @@ defmodule Backpex.Resource do
     |> maybe_join(associations)
     |> maybe_preload(associations, fields)
     |> maybe_merge_dynamic_fields(fields)
-    |> where([{^schema_name, schema_name}], schema_name.id == ^id)
-    |> repo.one()
+    |> where_id(schema_name, id)
+    |> repo.one!()
+  end
+
+  defp where_id(query, schema_name, id) do
+    case Ecto.UUID.cast(id) do
+      {:ok, valid_id} -> where(query, [{^schema_name, schema_name}], schema_name.id == ^valid_id)
+      :error -> where(query, [{^schema_name, schema_name}], true == false)
+    end
   end
```

Comment:
@Flo0807 This fixes another issue with invalid IDs when visiting `/admin/posts/INVALID/show` (for example) you would otherwise get an error like this:

```
/opt/app/lib/backpex/resource.ex:273: value `"INVALID"` cannot be dumped to type :binary_id in query:

from p0 in Demo.Post,
  as: :post,
  left_join: u1 in Demo.User,
  as: :user,
  on: p0.user_id == u1.id,
  left_join: c2 in Demo.Category,
  as: :custom_category,
  on: p0.category_id == c2.id,
  where: p0.id == ^"INVALID",
  distinct: [asc: p0.id],
  select: p0,
  preload: [{:user, #Ecto.Query<from u0 in Demo.User, as: :user, select: merge(u0, %{full_name: fragment("concat(?, ' ', ?)", u0.first_name, u0.last_name)})>}, :category, :tags]
```

The error case query `true == false` will result in a `Ecto.NoResultsError` and properly render a 404 page.